dashboard/app/views/levels/editors/_blockly.html.haml

```diff
@@ -183,99 +183,119 @@
-      if ($('.qtip_position_adjust_x_' + i).val().length > 0)
-        updated_callouts[i].qtip_config.position.adjust.x = $('.qtip_position_adjust_x_' + i).val();
-      if ($('.qtip_position_adjust_y_' + i).val().length > 0)
-        updated_callouts[i].qtip_config.position.adjust.y = $('.qtip_position_adjust_y_' + i).val();
-      if ($(".qtip_hide_" + i).prop("checked") == true)
+    function save_callouts_in_updated_callouts(i) {
+      id = callout_spaces[i];
+      updated_callouts[i].localization_key = $('.callout_' + id + ' .localization_key').val();
+      updated_callouts[i].callout_text = $('.callout_' + id + ' .callout_text').val();
+      updated_callouts[i].element_id = $('.callout_' + id + ' .element_id').val();
+      updated_callouts[i].on = $('.callout_' + id + ' .on').val();
+      updated_callouts[i].qtip_config.position.my = $('.callout_' + id + ' .qtip_position_my').val();
+      updated_callouts[i].qtip_config.position.at = $('.callout_' + id + ' .qtip_position_at').val();
+      updated_callouts[i].qtip_config.position.adjust.x = parseInt($('.callout_' + id + ' .qtip_position_adjust_x').val());
+      updated_callouts[i].qtip_config.position.adjust.y = parseInt($('.callout_' + id + ' .qtip_position_adjust_y').val());
+      if ($('.callout_' + id + ' .qtip_hide').prop("checked") == true)
         updated_callouts[i].qtip_config.style.classes = "no-tip";
+      else
+        updated_callouts[i].qtip_config.style.classes = "";
     }
 
-    $(".callouteditor input, .callouteditor select").change(function() {
-      for(i=0; i<updated_callouts.length; i++) {
-        updated_callouts[i].localization_key = $(".localization_key_" + i).val();
-        updated_callouts[i].callout_text = $(".callout_text_" + i).val();
-        updated_callouts[i].element_id = $(".element_id_" + i).val();
-        updated_callouts[i].on = $(".on_" + i).val();
-        updated_callouts[i].qtip_config.position.my = $(".qtip_position_my_" + i).val();
-        updated_callouts[i].qtip_config.position.at = $(".qtip_position_at_" + i).val();
-        updated_callouts[i].qtip_config.position.adjust.x = parseInt($(".qtip_position_adjust_x_" + i).val());
-        updated_callouts[i].qtip_config.position.adjust.y = parseInt($(".qtip_position_adjust_y_" + i).val());
+    function add_empty_callout() {
+      var callout = {};
+      callout.localization_key = "";
+      callout.callout_text = "";
+      callout.element_id = "";
+      callout.on = "";
+      callout.qtip_config = {};
+      callout.qtip_config.position = {};
+      callout.qtip_config.position.adjust = {};
+      callout.qtip_config.style = {};
+      updated_callouts.push(callout);
+    }
 
-        if ($(".qtip_hide_" + i).prop("checked") == true)
-          updated_callouts[i].qtip_config.style.classes = "no-tip";
-        else
-          updated_callouts[i].qtip_config.style.classes = "";
+    function create_new_callout_space() {
+      var callout_html = $("#callout_template").html();
```

Comment:
Prefer using jQuery's [`.clone()`](http://api.jquery.com/clone/) and setting the class via `.addClass()`.